# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1423174367 -3600
#      Thu Feb 05 23:12:47 2015 +0100
# Node ID a9516bc4c55c161047d0910ca7f72252a9163c1f
# Parent  5a59c0e1203da8ca08a1fff461ffdfa877c09a93
Update axes "tightinset" when "x/y/ztick" have changed

* graphics.cc (axes::properties::update_x/y/ztick): call sync_positions to update the tightinset.

* graphics.cc (axes::properties::set_x/y/zticklabel): call sync_positions to update the tightinset.

diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -6391,48 +6391,54 @@ axes::properties::set_xticklabel (const 
       if (xticklabel.set (convert_ticklabel_string (v), false))
         {
           set_xticklabelmode ("manual");
           xticklabel.run_listeners (POSTSET);
           mark_modified ();
         }
       else
         set_xticklabelmode ("manual");
+
+      sync_positions ();
     }
 }
 
 void
 axes::properties::set_yticklabel (const octave_value& v)
 {
   if (!error_state)
     {
       if (yticklabel.set (convert_ticklabel_string (v), false))
         {
           set_yticklabelmode ("manual");
           yticklabel.run_listeners (POSTSET);
           mark_modified ();
         }
       else
         set_yticklabelmode ("manual");
+
+      sync_positions ();
     }
 }
 
 void
 axes::properties::set_zticklabel (const octave_value& v)
 {
   if (!error_state)
     {
       if (zticklabel.set (convert_ticklabel_string (v), false))
         {
           set_zticklabelmode ("manual");
           zticklabel.run_listeners (POSTSET);
           mark_modified ();
         }
       else
         set_zticklabelmode ("manual");
+
+      sync_positions ();
     }
 }
 
 // Almost identical to convert_ticklabel_string but it only accepts
 // cellstr or string, not numeric input.
 static octave_value
 convert_linestyleorder_string (const octave_value& val)
 {
diff --git a/libinterp/corefcn/graphics.in.h b/libinterp/corefcn/graphics.in.h
--- a/libinterp/corefcn/graphics.in.h
+++ b/libinterp/corefcn/graphics.in.h
@@ -3976,26 +3976,29 @@ public:
     void update_ticklength (void);
     void update_tickdir (void) { update_ticklength (); }
     void update_tickdirmode (void) { update_ticklength (); }
 
     void update_xtick (void)
     {
       if (xticklabelmode.is ("auto"))
         calc_ticklabels (xtick, xticklabel, xscale.is ("log"));
+      sync_positions ();
     }
     void update_ytick (void)
     {
       if (yticklabelmode.is ("auto"))
         calc_ticklabels (ytick, yticklabel, yscale.is ("log"));
+      sync_positions ();
     }
     void update_ztick (void)
     {
       if (zticklabelmode.is ("auto"))
         calc_ticklabels (ztick, zticklabel, zscale.is ("log"));
+      sync_positions ();
     }
 
     void update_xtickmode (void)
     {
       if (xtickmode.is ("auto"))
         {
           calc_ticks_and_lims (xlim, xtick, xmtick, xlimmode.is ("auto"),
                                xscale.is ("log"));
